spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceHttpAutoConfiguration.java

```diff
@@ -138,3 +139,19 @@ HttpSampler sleuthClientSampler() {
 		return leftDecision && rightDecision;
 	}
 }
+
+class PathMatchingHttpSampler extends HttpSampler {
+	private final SleuthWebProperties properties;
+
+	PathMatchingHttpSampler(SleuthWebProperties properties) {
+		this.properties = properties;
+	}
+
+	@Override public <Req> Boolean trySample(HttpAdapter<Req, ?> adapter, Req request) {
+		String path = adapter.path(request);
+		if (path == null) {
+			return null;
+		}
+		return path.matches(this.properties.getClient().getSkipPattern()) ? false : null;
```

Comment:
If the skip pattern isn't set, does this NPE?

Reply:
Good point - I'll fix it to empty string